extensions/crypto/01_crypto.js

```diff
@@ -41,10 +42,83 @@
     return arrayBufferView;
   }
 
+  const supportedAlgorithms = {
+    "digest": {
+      "SHA-1": {},
+      "SHA-256": {},
+      "SHA-384": {},
+      "SHA-512": {},
+    },
+  };
+
+  function normalizeAlgorithm(algorithm, op) {
+    if (typeof algorithm == "string") {
+      return normalizeAlgorithm({ name: algorithm }, op);
+    }
+
+    const initialAlgorithm = webidl.converters["Algorithm"](algorithm, {
+      context: "Argument 1",
+    });
+
+    const registeredAlgorithms = supportedAlgorithms[op];
+    const algorithmName = Object.keys(registeredAlgorithms)
+      .find((key) => key.toLowerCase() == initialAlgorithm.name.toLowerCase());
```

Comment:
I'm not thrilled by how roundabout and complicated this approach is. Could simply be:
```js
// Should match op_crypto_subtle_digest() in extensions/crypto/lib.rs 
function digestToId(name) {
  switch (name) {
    case "SHA-1": return 0;
    case "SHA-256": return 1;
    case "SHA-384": return 2;
    case "SHA-512": return 3;
  }
  throw new DOMException(/* ... */);
}

name = name.toUpperCase();
const id = digestToId(name);
const normalizedAlgorithm = { name };
// etc.
```

Reply:
Initial pass was like that, now the whole normalisation thing follows the spec very literally